Assets/Mirage/Runtime/PlayerSpawner.cs

```diff
@@ -35,7 +35,15 @@ public virtual void Start()
             }
             if (Client != null)
             {
-                SceneManager.ClientSceneChanged.AddListener(OnClientSceneChanged);
+                if(SceneManager != null)
+                {
+                    SceneManager.ClientSceneChanged.AddListener(OnClientSceneChanged);
+                }
+                else
+                {
+                    Client.Authenticated.AddListener(c => Client.Send(new AddPlayerMessage()));
```

Comment:
will this event need to be removed in onDestroy?

Reply:
Hmm. It would be cleaned up when the network is destroyed likely a client shutdown. But it would add duplicates in a start/stop/start scenario. Is there a way to remove a listener written like that or do I need to move it to a normal method?